java/client/src/org/openqa/selenium/firefox/internal/Executable.java

```diff
@@ -123,22 +123,24 @@ private static File locateFirefoxBinaryFromSystemProperty() {
     if (binaryName == null)
       return null;
 
-    File binary = new File(binaryName);
-    if (binary.exists())
```

Comment:
if you did that, you wouldn't need to make the adjustments below to the mac things.